sys/arm64/include/vmparam.h

```diff
@@ -228,6 +228,9 @@
 	((vm_offset_t)(va) - (vm_offset_t)_DMAP_BASE) + dmap_phys_base;	\
 })
 
+#define	PHYS_TO_DMAP_PAGE(pa)					\
+	cheri_kern_setboundsexact(PHYS_TO_DMAP(pa), PAGE_SIZE)
+
 #define	VM_MIN_USER_ADDRESS	(0x0000000000000000UL)
```

Comment:
One thing you might also consider is a `PHYS_TO_DMAP_LEN` or some such that sets the bounds that you could use in several of the places you are still setting explicit bounds?  I would maybe make that wrapper macro assert that the [`pa`, `pa + size`) range does not span a page boundary though.